crates/proof-of-sql/src/base/scalar/mont_scalar.rs

```diff
@@ -356,6 +356,26 @@ impl<T: MontConfig<4>> Display for MontScalar<T> {
     }
 }
 
+macro_rules! mont_scalar_conversion_to_bigint {
+    ($scalar:ty) => {
+        impl From<$scalar> for BigInt {
+            fn from(value: $scalar) -> Self {
+                // Since we wrap around in finite fields anything greater than the max signed value is negative
+                let is_negative = value > <$scalar>::MAX_SIGNED;
+                let sign = if is_negative {
+                    num_bigint::Sign::Minus
+                } else {
+                    num_bigint::Sign::Plus
+                };
+                let value_abs: ark_ff::BigInt<4> =
+                    (if is_negative { -value } else { value }).0.into_bigint();
+                BigInt::from_bytes_le(sign, &value_abs.to_bytes_le())
+            }
+        }
+    };
+}
+pub(crate) use mont_scalar_conversion_to_bigint;
```

Comment:
NIT:  I'd prefer to minimize the code that goes in macros as they tend to obfuscate compilation issues. Does writing this conversion generically on `S: Scalar` cause conflicting implementations or something?

If so, could you instead write a generic `fn scalar_to_bigint<S: Scalar>` that has the bulk of this LOC and is called by the macros?

Reply:
@tlovell-sxt This code actually requires to use implementation details of both `MontScalar` and `Scalar::MAX_SIGNED`. Hence the macro is needed.

Reply:
You should be able to use `Into<[u64; 4]>` rather than the `MontScalar` specific things.

Reply:
@JayWhite2357 Yup! Done.